src/usePager.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { useContext, useSyncExternalStore } from 'react';
+
+export type PagerViewContextValue = {
+  store: PagerStore | null;
+  setPage: (selectedPage: number) => void;
+  setPageWithoutAnimation: (selectedPage: number) => void;
+  setScrollEnabled: (scrollEnabled: boolean) => void;
+};
+
+export type PagerState = {
+  page: number;
+  hasNextPage: boolean;
+  hasPreviousPage: boolean;
+};
+
+export type PagerStore = {
+  getState: () => PagerState;
+  setState: (state: PagerState) => void;
+  subscribe: (listener: () => void) => () => void;
+};
+
+export const PagerViewContext =
+  React.createContext<PagerViewContextValue | null>(null);
+
+export const createPagerStore = (initialPage: number) => {
+  let state: PagerState = {
+    page: initialPage,
+    hasNextPage: false,
+    hasPreviousPage: false,
+  };
+
+  const getState = () => {
+    return state;
+  };
+
+  const listeners = new Set<() => void>();
+
+  const emitChange = () => {
+    for (const listener of listeners) {
+      listener();
+    }
+  };
+
+  const setState = (newState: PagerState) => {
+    state = newState;
+    emitChange();
+  };
+
+  const subscribe = (listener: () => void) => {
+    listeners.add(listener);
+    return () => listeners.delete(listener);
+  };
+
+  return { getState, setState, subscribe };
+};
+
+export const usePager = () => {
+  if (Number(React.version.split('.')[0]) < 18) {
+    throw new Error('usePager requires React 18 or later.');
+  }
```

Comment:
If your library users are below react 18 and you don't have `useSyncExternalStore`, you can install `use-sync-external-store` for backwards compatibility and add `"dependencies"` inside this project's package.json (zustand already does that).

Currently, `react-native-pager-view` is a zero-dependency. However, since the project doesn't emphasize zero-dependency, it already has a dependency on react, and finally, `use-sync-external-store` is an official library, it doesn't seem to be a problem.